app/scripts/lib/fxa-client.js

```diff
@@ -104,7 +104,21 @@ function (_, FxaClient, $, xhr, p, Session, AuthErrors, Constants) {
 
       return self._getClient()
         .then(function (client) {
-          return client.signIn(email, password, { keys: relier.wantsKeys() });
+          var signInOptions = {
+            keys: relier.wantsKeys()
+          };
+
+          // `service` is sent on signIn to notify users when a new service
+          // has been attached to their account. Notifications should only
+          // be sent when the user is signing in to a site. signIn is
+          // also called after a password change or reset. The `sendService`
+          // flag is passed in when the service should be sent (when the
+          // user is signing in).
+          if (relier.has('service') && options.sendService) {
```

Comment:
This is my one concern about this PR. Using `options.sendService` feels gross. The reason I do this is because signIn is also called after a password change and reset. The only time the service should be sent is after an actual signIn. I looked at having the `changePassword` and `completePasswordReset` calls call `signIn`, but that makes the signature of both functions expand to take both a relier and options block. 

Any suggestions would be helpful.